src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1465,6 +1465,12 @@
       (format "ALTER TABLE reports ADD COLUMN logs %s DEFAULT NULL" jsonb-type)
       (format "ALTER TABLE reports ADD COLUMN resources %s DEFAULT NULL" jsonb-type))))
 
+(defn factset-hash-field-not-nullable
+  []
+  (jdbc/do-commands
+   "UPDATE factsets SET hash='' WHERE hash is NULL"
```

Comment:
I think we'd be fine with a pk as a string for that value. It'll be updated on next facset anyway. Why I am asking is we could have always done that, there's nothing 4.0.0 specific about that.